index.js

```diff
@@ -2,45 +2,66 @@ const Entity = require('siren-parser');
 const Action = require('siren-parser/dist/Action');
 const Link = require('siren-parser/dist/Link');
 
+/**
+ * siren-parser wrapper that supports following links and performing actions
+ */
 class Client {
   /**
    * performs an async HTTP request
    * returns object { body, contentType }
    *  body: response body
    *  contentType: Content-Type response header
    *
-   * @callback request
-   * @param {string} uri (required) uri of the resource
+   * @typedef {Object} Response
+   * @property body response body
+   * @property contentType Content-Type response header
+   *
+   * @callback requestFn
+   * @param {string} href (required) href of the resource
    * @param {string} method (required) HTTP request verb ('GET', 'POST', 'PUT', 'PATCH', 'DELETE')
-   * @param {object} fieldValues (optional) field names and values for the request
-   * @return {object} body: response body, contentType: Content-Type response header
+   * @param {Object} [fieldValues] (optional) field names and values for the request
+   * @return {Response} body: response body, contentType: Content-Type response header
    */
 
   /**
    * Setup the fluent siren client
    *
-   * @param {request} requestFn performs an async HTTP request
+   * @typedef {Object} Options
+   * @property {string|string[]} [originWhitelist] (optional) Whitelist of allowed origins (e.g. 'https://example.com')
+   *
+   * @param {requestFn} requestFn (required) performs an async HTTP request
+   * @param {Options} [options] (optional) configuration options for the client
    */
-  constructor(requestFn) {
+  constructor(requestFn, options = {}) {
     this._requestFn = requestFn;
+    this._options = options;
   }
 
+  /**
+   * Starts a client by fetching the provided href
+   * @param {string} href
+   * @returns {Promise<Entity>}
+   */
   async start(href) {
     return this.perform(href, 'GET');
   }
 
+  /**
+   * Performs a request
+   * @param {string} href
+   * @param {string} method
+   * @param {Object} [fieldValues]
+   * @returns {Promise<Entity>}
+   */
   async perform(href, method, fieldValues) {
-    const { body, contentType } = await this._requestFn(href, method, fieldValues);
-    if (!body || !contentType || !_isSiren(contentType)) {
-      return body;
-    }
-
-    const entity = new Entity(body);
-    this.attachClient(entity);
-
-    return entity;
+    _verifyOrigin(href, this._options.originWhitelist);
```

Comment:
I don't think this should verify the origin on every call - JUST the start call and let it perform or follow any link after that.

Reply:
yes, that's exactly what is happening

Reply:
only start and the first perform will do the verification